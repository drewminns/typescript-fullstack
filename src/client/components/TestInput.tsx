import * as React from 'react';

export interface ITestInputState {
  name: string;
}

export class TestInput extends React.Component<{}, ITestInputState> {
  state = {
    name: '',
  };

  handleInput = () => {
    this.setState({ name: 'drew' });
  };

  handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    e.preventDefault();
    this.setState({ name: e.target.value });
  };

  render() {
    return (
      <div>
        <input type="text" value={this.state.name} onChange={this.handleChange} aria-label="test-input" />
        <button onClick={this.handleInput}>Get New Name</button>
      </div>
    );
  }
}
