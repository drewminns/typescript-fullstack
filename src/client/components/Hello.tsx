import * as React from 'react';
import { TestInput } from './TestInput';
export interface IHelloProps {
  compiler: string;
  framework: string;
}

// 'HelloProps' describes the shape of props.
// State is never set so we use the '{}' type.
export class Hello extends React.Component<IHelloProps, {}> {
  render() {
    return (
      <div>
        <h1 data-testid="string-data">
          Hello from {this.props.compiler} and {this.props.framework}!
        </h1>
        <TestInput />
      </div>
    );
  }
}
