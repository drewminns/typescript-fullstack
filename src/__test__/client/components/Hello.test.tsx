import * as React from 'react';
import { Hello } from '../../../client/components/Hello';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

describe('components/Hello', () => {
  const MOCK = { compiler: 'TypeScript', framework: 'React' };
  it('Hello mounts properly', () => {
    const { container } = render(<Hello compiler={MOCK.compiler} framework={MOCK.framework} />);
    expect(container).toMatchSnapshot();
  });

  it('Hello mounts properly', () => {
    const { getByTestId } = render(<Hello compiler={MOCK.compiler} framework={MOCK.framework} />);
    expect(getByTestId('string-data')).toHaveTextContent(`Hello from ${MOCK.compiler} and ${MOCK.framework}!`);
  });
});
