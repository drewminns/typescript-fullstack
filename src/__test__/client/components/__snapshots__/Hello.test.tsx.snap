// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Hello Hello mounts properly 1`] = `
<div>
  <div>
    <h1
      data-testid="string-data"
    >
      Hello from 
      TypeScript
       and 
      React
      !
    </h1>
    <div>
      <input
        aria-label="test-input"
        type="text"
        value=""
      />
      <button>
        Get New Name
      </button>
    </div>
  </div>
</div>
`;
